lib/usd/hdMaya/delegates/delegate.h

```diff
@@ -124,6 +128,55 @@ class HdMayaDelegate
     inline HdEngine&          GetEngine() { return _engine; }
     inline HdxTaskController* GetTaskController() { return _taskController; }
 
+    /// Calls that mirror UsdImagingDelegate
+
+    /// Setup for the shutter open and close to be used for motion sampling.
+    HDMAYA_API
+    void SetCameraForSampling(SdfPath const& id);
+
+    /// Returns the current interval that will be used when using the
+    /// sample* API in the scene delegate.
+    HDMAYA_API
+    GfInterval GetCurrentTimeSamplingInterval() const;
+
+    /// Common function to return templated sample types 
+    template <typename T, typename Getter> HDMAYA_API
+    size_t SampleValues(size_t maxSampleCount, float* times, T* samples, Getter getValue)
+    {
+        if (ARCH_UNLIKELY(maxSampleCount == 0)) {
+            return 0;
+        }
+        if (maxSampleCount == 1 || !GetParams().enableMotionSamples()) {
```

Comment:
Feel like this is technically wrong behavior for `maxSampleCount == 1` if, say, `startTime = endTime = .25`.  Instead,  I think we should just set `tStep = 0` if `maxSampleCount == 1`, and otherwise I think the "normal" code branch should work.

Reply:
I'm fine with that, but perhaps @kxl-adsk can clarify, is there any overhead to the `MDGContextGuard` for the case where we can just be querying maya value's directly ?
i.e Should we keep an optimized case of 1 sample where `startTime == 0` vs the below
```
MDGContextGuard guard(<CURRENT_MAYA_TIME>);
...Query Maya Values...
```

Reply:
Keeping the _fast path_ for one sample where `sampleTime=currentFrame` for now, should be easy to delete the lines if they proove to be irrelevant.

Also in the case of one-sample, I'm using the center of the shutter-range so 1 sample for [-0.25, -0.25] and [-1, 1] would both be **t=0**.  It seems a bit more intuitive to me.

Reply:
>  is there any overhead to the MDGContextGuard for the case where we can just be querying maya value's directly ?

Each node has it state and each state is associated with a context. When you start a new context computation, you will have to compute the state for it. We don't assume that what you need are results from normal context (current time). Reading values from the current time should happen within a normal context.

Reply:
Thanks.  The fast-path of 1 sample at t=0 is still there.